### DISCUSSION TITLE
Accessing StateObject's object without being installed on a View.

### DISCUSSION THREAD

**Discussion by pobengtsson** [nalexn/ViewInspector#152] [Q&A]
Hi all,

What do you do when you run your tests and get logging like this in the console:

```
[SwiftUI] Accessing StateObject's object without being installed on a View. This will create a new instance each time.
[SwiftUI] Accessing State's value outside of being installed on a View. This will result in a constant Binding of the initial value and will not update.
```

I am pretty sure not all these printouts does not show up when running the app, and I was trying to fix my viewinspector based tests so that I got less of them.

I find, that when I use find/findAll, these printouts/warnings show up. and when I use stepwise navigation to find subviews, the warnings mostly disappear.

I want to use find/findAll to make my tests less sensitive to UI changes.
But I do not want the printout of tests to have (possibly? false) warnings about potential errors, as that lowers the signal to noise ration and real problems will slip through.

I would appreciate to learn from how others deal with this?
Is it a problem, that nobody else experience at all?

**Comment by cds-nl** [comment]
Hi :) 
I'm I have the same issue, did you managed to find a workaround ? 
Thank you.

**Comment by cds-nl** [comment]
Hi everyone 
One of the solution is to use the view reference from the didApper callback (see example in the documentation)
You're going to have an error shown in xocde, but you'll be able to save the view reference and interact with it

**Comment by sisoje** [comment]
I have the same issue.
ViewInspector tries to copy the body, but the StateObject is retained by the original view so it copies the view without original StateObject.
When I put ObservedObject in my code then there is no warning.
